src/Navigation/NavigationItem.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { NavLink as L, useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import { StyledIcon } from '@styled-icons/styled-icon';
+import { Paragraph as P } from '../Text';
+import { Mixins } from '../Utils';
+
+const hydratePath = (path: string, params: NavigationParams) =>
+    path.replace(/(:[0-9A-Za-z]*)/g, (e) => params[e.slice(1)]);
+
+interface NavigationIconProps {
+    icon?: StyledIcon;
+    to?: string;
+    type?: any | string | number | symbol;
+}
+interface NavigationParams {
+    id: string;
+}
+
+export const NavigationItem: React.FC<NavigationIconProps> = ({
+    children,
+    icon,
+    to = '',
+    type,
+    ...props
+}) => {
+    const params = useParams<NavigationParams>();
+    const isExternal = to.startsWith('http');
+    return (
+        <Item as={type} {...props}>
+            <NavLink
+                to={hydratePath(to, params)}
+                target={isExternal ? '_blank' : ''}
```

Comment:
This being hard coded is not congruent with the React-UI library. It should be extensible, is there a way we can effect the same logic but have it handled externally?

Reply:
This is for open route in a new page when it is external ('http...' something). It depends on the prop to passed to the component. So why I should handle this externally?

Reply:
React-UI library components should handle how the component looks an not the logic. If they handle logic then it limits their extensibility.

Reply:
bump